Document how to debug the Hexatomic product from Eclipse. (#52)
hexatomic/hexatomic: docs/dev/src/development/debug-with-eclipse.md

# Debug with Eclipse

Hexatomic is a modular project and while it is possible to debug a user-defined sub-set of the bundles,
in most cases debugging the complete application is more useful.

To debug or start Hexatomic with all the features and bundles, you have to open the  `org.corpus_tools.hexatomic.product` project which contains the Hexatomic product definition.

> In this project open the `org.corpus_tools.hexatomic.product` file in Eclipse.

Reviewer: ```suggestion
In this project, open the `org.corpus_tools.hexatomic.product` file in Eclipse.
```

;)

![Location of the product definition file in the Project Explorer](./product-file-location.png)

> To debug Hexatomic, click on the `Launch an Eclipse application in Debug mode` link on the overview tab of the product configuration file editor.

Reviewer: ```suggestion
To debug Hexatomic, click on the `Launch an Eclipse application in Debug mode` link in the overview tab of the Product Configuration Editor.
```

> To just run the product, but not debug it, choose `Launch an Eclipse application` on the same page.

Reviewer: ```suggestion
To run the product without debugging, click `Launch an Eclipse application` on the same page.
If you don't see the Product Configuration Editor, you may have opened the file with a different editor instead.
In this case, right-click on the `org.corpus_tools.hexatomic.product` file, and choose *Open With* > *Product Configuration Editor*.
```
